.github/workflows/libosrm.yml

```
@@ -0,0 +1,51 @@
+name: Libosrm CI
+on:
+  push:
+     branches:
+       - master
+     paths:
+       - '.github/workflows/libosrm.yml'
+       - '**libosrm_wrapper**'
+  pull_request:
+     branches:
+       - master
+     paths:
+       - '.github/workflows/libosrm.yml'
+       - '**libosrm_wrapper**'
+jobs:
+  libosrm:
+    strategy:
+      matrix: 
+        cxx: ['g++-9', 'clang++-10']
+    runs-on: ubuntu-20.04
```

@Orthae GA makes running the workflow in your own image pretty easy, this should be enough:

```suggestion
    container: osrm/osrm-backend:v5.25.0
```

Then you should be able to simply remove the OSRM parts here and jump directly to compiling vroom in some directory.

---

I checked it, and it seems that the entire pipeline has to be running inside the osrm container, and there will need for manually installing all dependencies. Also, the image is on Debian 9 which, gives access to g++6.

What about cache? Adding osrm to Github Actions cache after compiling, will help with subsequent runs when someone actively works on something related to libosrm.

---

> gives access to g++6.

hmpf yeah that might be too old for vroom, not sure.. 

If you think cache is easy to implement here, sure.. I'm not very versed with caching builds, but could imagine that g++/clang++ matrix could create problems? Do they cache well? I usually use ccache, might be easier here as well?

---

Disclaimer, I'm not in the vroom core team, so this is "just" a suggestion to save some CI hours. 

you're already restricting it to the few paths that are triggering this and the travis jobs didn't have any cache either. implementing cache might take more effort than it's worth maybe..

---

Added cache, subsequent build are much faster now.

---

This cache thing is really neat. It will be much easier for us to check build against `libosrm` as it will only be built once (per version).